jerry-core/lit/lit-char-helpers.cpp

```diff
@@ -253,36 +253,78 @@ lit_char_is_unicode_connector_punctuation (ecma_char_t c) /**< code unit */
 } /* lit_char_is_unicode_connector_punctuation */
 
 /**
- * Check if specified character is a word character (part of IsWordChar abstract operation)
+ * Check if specified character is one of OctalDigit characters (ECMA-262 v5, B.1.2)
  *
- * See also: ECMA-262 v5, 15.10.2.6 (IsWordChar)
+ * @return true / false
+ */
+bool
+lit_char_is_octal_digit (ecma_char_t c) /**< code unit */
+{
+  if (c >= LIT_CHAR_ASCII_OCTAL_DIGITS_BEGIN && c <= LIT_CHAR_ASCII_OCTAL_DIGITS_END)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+} /* lit_char_is_octal_digit */
+
+/**
+ * Check if specified character is one of DecimalDigit characters (ECMA-262 v5, 7.8.3)
  *
- * @return true - if the character is a word character
- *         false - otherwise.
+ * @return true / false
  */
 bool
-lit_char_is_word_char (ecma_char_t c) /**< code unit */
+lit_char_is_decimal_digit (ecma_char_t c) /**< code unit */
```

Comment:
@sand1k , `Date.parse` is on master now and it's using `lit_char_is_unicode_digit`. If you introduce `lit_char_is_decimal_digit` in this patch, then could you change the `lit_char_is_unicode_digit` call in `ecma-builtin-date.cpp` to `lit_char_is_decimal_digit` in your patch too?